Row.c code shrink and adding a new format for Row_printNanoseconds()

Explorer09 opened the pull request:
The first three commits are code shrinks.
The last one commit adds a new format for Row_printNanoseconds() because I feel I missed a chance of allowing one extra digit of precision to be printed.
The last commit is optional. If people feel the new format is not worth it, I can drop that commit.

(Follow-up of #1425)

Explorer09 committed:
Row_printKBytes() code shrink

* Reduce length of the format strings by removing redundant field
  width specifications (e.g. "%1u" -> "%u").
* Merge some identical function calls in conditionals.

Signed-off-by: Kang-Che Sung <explorer09@gmail.com>

Explorer09 committed:
Row_printTime() code shrink

Shorten a format string ("%1ud" -> "%ud").

Signed-off-by: Kang-Che Sung <explorer09@gmail.com>

BenBE reviewed (approved):
LGTM. Although a bit unconventional to skip from `ns` to `ms` right away … Also, even though we have Unicode, for units I'd still prefer to use `us` unconditionally.

Explorer09 commented:
> LGTM. Although a bit unconventional to skip from `ns` to `ms` right away … Also, even though we have Unicode, for units I'd still prefer to use `us` unconditionally.

"1.0000ms" vs. "1000.0us" ...

These are same precision for showing the fractions of a second. I don't know which one looks better.

Note that the time units in the GPUMeter would have to use `us` anyway (due to shorter width of the field). I have thought of adding the code to display `μs` there.

BenBE commented:
I'm neutral either way …

fasterit commented:
Can we have one "format times" code as in #1583 and this combined?

fasterit commented:
https://en.wikipedia.org/wiki/Microsecond ... "Its symbol is μs, sometimes simplified to us when Unicode is not available."
No need to ignore the unit. It's 2025 and we have `#ifdef HAVE_LIBNCURSESW` :tada: .

Explorer09 commented:
> Can we have one "format times" code as in #1583 and this combined?

The output of #1583 is an ASCII string (`char []`), potentially allowing UTF-8.
The output of this one (#1579 and #1425) is a `RichString` buffer.
So, nope. While I wish the two routines be merged, it's impossible due to the difference of output object types.

> https://en.wikipedia.org/wiki/Microsecond ... "Its symbol is μs, sometimes simplified to us when Unicode is not available." No need to ignore the unit. It's 2025 and we have `#ifdef HAVE_LIBNCURSESW` 🎉 .

Ask @BenBE for this. Actually I wished the Greek letter mu (μ, U+03BC) be used when it's available, but [a previous comment of BenBE](https://github.com/htop-dev/htop/pull/1579#pullrequestreview-2528158457) said something about using `us` unconditionally...

BenBE commented:
Using `us` instead of `µs` was suggested in my comment mostly for two reasons:
1. Avoids distinguishing cases in the code whether Unicode is available or not.
2. Avoids issues with printing on the terminal when locales are badly configured.

At least the second situation can occur easily when working with `screen` and `tmux` over SSH when some of the environment variables are forwarded incorrectly or mixed up due to diverging settings on different computers. Using `us`, while not being canonical, avoids both these issues and is usually understood just as well.

@fasterit: For me it's not a pressing issue and I'm totally indifferent about which one we use in the end.

Explorer09 commented:
> Using `us` instead of `µs` was suggested in my comment mostly for two reasons:
> 
> 1. Avoids distinguishing cases in the code whether Unicode is available or not.
> 2. Avoids issues with printing on the terminal when locales are badly configured.
> 
> At least the second situation can occur easily when working with `screen` and `tmux` over SSH when some of the environment variables are forwarded incorrectly or mixed up due to diverging settings on different computers. Using `us`, while not being canonical, avoids both these issues and is usually understood just as well.
> 

I'm personally neutral on this, because after #1598 it would be easier to add the μ symbol support while having `u` as an appropriate fallback. It's just extra code that some people might consider it not worth it.

Explorer09 committed:
Row_printNanoseconds() code shrink

Merge two xSnprintf() calls within the function. When the time value to
print is less than one second, the "%.u" format allows us to print no
digits in the seconds field.

Signed-off-by: Kang-Che Sung <explorer09@gmail.com>

Explorer09 committed:
Add a new printing format to Row_printNanoseconds()

Add a new format: "1.0000ms" - "9.9999ms".
Previously they would print as ".001000s" and ".009999s", and this new
format adds one extra digit of precision.
Note that I print the unit as "ms" rather than microseconds; this saves
code for deciding whether to print the Greek "mu" or the Latin "u".

Row_printNanoseconds() now prints this list of formats:
"     0ns", "999999ns", "1.0000ms", "9.9999ms", ".010000s", ".999999s",
"1.00000s", "59.9999s", "1:00.000", "9:59.999", "10:00.00" ...

Signed-off-by: Kang-Che Sung <explorer09@gmail.com>
